tests/test_person.py

```diff
@@ -50,6 +50,29 @@ def test_edit_person_mac(client):
     assert person.last_name == 'l_test'
     assert person.first_name == 'f_test'
 
+
+def test_get_delete_person_mac(client):
+    resp = client.get(
+        url_for('api.delete_mac', mac_address='00 00 00 00 00 00'))
+    assert resp.status_code == 200
+
+
+def test_delete_person_mac(client):
+    person = PersonFactory()
+    resp = client.post(url_for('api.add_mac'),
+                       dict(mac='00:00:00:00:00:00', device='mobile',
+                            person=1))
+    assert resp.status_code == 200
+    assert len(Address.query.all()) == 1
+    address = Address.query.first()
+    assert address.deleted is False
+
+    resp = client.post(url_for('api.delete_mac', mac_address='00 00 00 00 00 00'))
+    assert resp.status_code == 200
+    address = Address.query.first()
+    assert address.deleted is True
+
```

Comment:
Try to write tests which test as little functionality as possible. Generate an Address with factory, generate a Person with factory who has that address and then test only the delete function.